nexus_constructor/component.py

```diff
@@ -1,13 +1,83 @@
 import attr
-from nexus_constructor.transformations import Transformation
+import h5py
+from typing import Any, List
 from nexus_constructor.pixel_data import PixelData
 from nexus_constructor.geometry_types import Geometry
-from typing import List
+from nexus_constructor.nexus import nexus_wrapper as nx
+from nexus_constructor.transformations import Transformation, create_transform
+
+
+class ComponentModel:
+    """
+    Provides an interface to an existing component group in a NeXus file
+    """
+
+    def __init__(self, nexus_file: nx.NexusWrapper, group: h5py.Group):
+        self.file = nexus_file
+        self.group = group
+
+    @property
+    def name(self):
+        return nx.get_name_of_group(self.group)
+
+    @name.setter
+    def name(self, new_name: str):
+        self.file.rename_group(self.group, new_name)
+
+    def get_field(self, name: str):
+        return self.file.get_field_value(self.group, name)
+
+    def set_field(self, name: str, value: Any, dtype=None):
+        self.file.set_field_value(self.group, name, value, dtype)
+
+    @property
+    def nx_class(self):
+        return self.file.get_nx_class(self.group)
+
+    @nx_class.setter
+    def nx_class(self, nx_class: str):
+        self.file.set_nx_class(self.group, nx_class)
+
+    @property
+    def description(self):
+        return self.file.get_field_value(self.group, "description")
+
+    @description.setter
+    def description(self, description: str):
+        if description:
+            self.file.set_field_value(self.group, "description", description, str)
+
+    @property
+    def transforms(self):
```

Comment:
For creating the UI, making the list of transformations hold a pointer to the component makes things a lot easier for me. See the [simple implementation](https://github.com/ess-dmsc/nexus-constructor/blob/add_component_view/nexus_constructor/component.py) that I have used for reference.

Reply:
We also need to figure out what transforms your interface should actually display. The list returned here is the full depends_on chain for this component. Only some, or none, of those transformations may live in the component group in the file.